### PR TITLE
[CI] Enable Android AVD cache

### PR DESCRIPTION
**Pull request by Gordon-F**
Enable AVD cache as discussed in #133. I hope it will fix problems with CI. Also includes changes from #198.

### CODE REVIEW

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -108,14 +108,37 @@ jobs:
...
         rustup target add x86_64-linux-android
+    - name: AVD cache
+      uses: actions/cache@v2
+      id: avd-cache
+      with:
+        path: |
+          ~/.android/avd/*
+          ~/.android/adb*
+        # One cache for 3 apk files
+        key: avd-emul
```
Comment (MarijnS95) [inline .github/workflows/rust.yml:119]: No need to mention that this is for 3 apk files specifically, just that this key is not variable meaning it doesn't depend on the job type/target... Doesn't need a comment at all IMO.

Perhaps `avd-emulator` reads nicer though?

File: .github/workflows/rust.yml [hunk 2]
```diff
@@ -108,14 +108,37 @@ jobs:
...
+    - name: create AVD and generate snapshot for caching
+      if: steps.avd-cache.outputs.cache-hit != 'true'
+      uses: reactivecircus/android-emulator-runner@v2.19.1
```
Comment (MarijnS95) [inline .github/workflows/rust.yml:122]:
```suggestion
      uses: reactivecircus/android-emulator-runner@v2
      
```

File: .github/workflows/rust.yml [hunk 3]
```diff
@@ -108,14 +108,37 @@ jobs:
...
+      with:
+        api-level: 29
+        arch: x86_64
+        profile: Nexus 6
+        target: default
+        force-avd-creation: true
+        emulator-options: -no-window -gpu swiftshader_indirect -noaudio -no-boot-anim -camera-back none
+        disable-animations: false
```
Comment (MarijnS95) [inline .github/workflows/rust.yml:130]: I saw this in the example, but I still don't understand why AVD creation should _NOT_ disable animations?

Comment (MarijnS95) [inline .github/workflows/rust.yml:130]: Let's keep this as similar as possible to the actual use:
```suggestion
        api-level: 29
        arch: x86_64
        # the `googleapis` emulator target is considerably slower on CI.
        target: default
        profile: Nexus 6
        disable-animations: true # Still not sure
        emulator-options: -no-window -gpu swiftshader_indirect -noaudio -no-boot-anim -camera-back none
        force-avd-creation: true
```

**Comment by MarijnS95** [comment]
> I had the same locally

With all these review comments I should have submitted my own approach instead :thinking: - but still thanks for doing the prep as well.

File: .github/workflows/rust.yml (hunk 3 above)
Comment (Gordon-F) [reply .github/workflows/rust.yml:130]: I don't know too :)

**Comment by Gordon-F** [comment]
> With all these review comments I should have submitted my own approach instead 🤔

Ok, let's close this in favor of your PR.

**Pull request closed without merging**